storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -173,6 +178,18 @@ public SafeFuture<Optional<SignedBlockAndState>> getSignedBlockAndStateInEffectA
    */
   public SafeFuture<List<BlobSidecar>> getBlobSidecars(
       final UInt64 slot, final List<UInt64> indices) {
+    if (spec.atSlot(slot).getMilestone().isGreaterThanOrEqualTo(SpecMilestone.EIP7594)) {
+      // TODO: suboptimal
```

Comment:
Do you think we need incorporate this complexity to the `CombinedChainDataClient` ? 
May be pull to the upper level classes?